utils/utils.ts

```diff
@@ -9,3 +10,26 @@ export const isValidPubKey = (key: string) => {
 
   return null
 }
+
+export const walletHasSplToken = async (connection, pubKey: string, splToken: string) => {
+  let accounts = await connection.getParsedProgramAccounts(
+      TOKEN_PROGRAM_ID,
+      {
+        filters: [
+          {
+            dataSize: 165,
+          },
+          {
+            memcmp: {
+              offset: 32,
+              bytes: pubKey, // base58 encoded string
+            },
+          },
+        ],
+      }
+  );
```

Comment:
I'd recommend using [getParsedTokenAccountsByOwner](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getParsedTokenAccountsByOwner) instead as `getParsedProgramAccounts` is slow and as far as I know doesn't currently work on mainnet.


```typescript
const accounts = (
  await connection.getParsedTokenAccountsByOwner(new PublicKey(pubKey), {
    programId: TOKEN_PROGRAM_ID,
  })
).value.map((account) => account.account.data.parsed.info.mint)
                                                                          
return accounts.includes(splToken)
```